src/utils/regexp.ts

```diff
@@ -1,5 +1,33 @@
+import { Context } from "../api";
 import { assert } from "./errors";
 
+
+/**
+ * Wrapper around RegExp.exec that takes into account the user's smart case
+ * preference.
+ */
+export function smartExec(re: RegExp, string: string): RegExpExecArray | null {
+  const pattern = re.source;
+  const flags = re.flags;
+
+  if (!Context.current.extension.isSmartCaseEnabled) {
+    return re.exec(string);
+  }
+
+  const hasUpperCase = /[A-Z]/.test(pattern);
+  if (hasUpperCase) {
+    return re.exec(string);
+  }
+
+  const adjustedFlags = flags.includes('i')
+    ? flags
+    : flags + 'i';
+
+  const adjustedRegex = new RegExp(pattern, adjustedFlags);
```

Comment:
nit: use double quotes, avoiding creating a new `RegExp` unless necessary

```suggestion
  const adjustedRegex = flags.includes("i") ? re : new RegExp(pattern, flags + "i");
```